src/redux/notifications/actions.ts

```diff
@@ -1,3 +1,180 @@
-// actions consts
-export const READ_NOTIFICATION = 'READ_NOTIFICATION';
-// export const DELETE_NOTIFICATION = 'DELETE_NOTIFICATION';
+import { HttpClient } from '../../core/utils/http-client';
+import CONFIG from '../../config';
+import { Notifications } from '../../core/messaging/notifications/notifications';
+import { captureException as SentryCaptureException } from '@sentry/react-native';
+import { getSelectedWallet } from '../wallets/selectors';
+import { Dispatch } from 'react';
+import { IReduxState } from '../state';
+import { PushNotifTokenType } from '../../core/messaging/types';
+import { IWalletState, IAccountState } from '../wallets/state';
+import { ChainIdType } from '../../core/blockchain/types';
+import { getTokenConfig } from '../tokens/static-selectors';
+
+export const SET_HAS_UNSEEN_NOTIFICATIONS = 'SET_HAS_UNSEEN_NOTIFICATIONS';
+export const SET_NOTIFICATIONS = 'SET_NOTIFICATIONS';
+export const MARK_SEEN = 'MARK_SEEN';
+
+const setHasUnseenNotifications = (hasUnseenNotifications: boolean) => {
+    return {
+        type: SET_HAS_UNSEEN_NOTIFICATIONS,
+        data: { hasUnseenNotifications }
```

Comment:
instead of having hasUnseenNotifications as a boolean, maybe we should have unseenNotifications as a number. it may be easier on the UI